### DISCUSSION TITLE
Solid bindings

### DISCUSSION THREAD

**Discussion by wobsoriano** [pmndrs/valtio#480] [Show and tell]
Published a solid primitive that uses `valtio/vanilla` - https://github.com/wobsoriano/solid-valtio

Usage:

```ts
import { proxy, useSnapshot } from 'solid-valtio'

const state = proxy({ count: 0 })

function Counter() {
  const snap = useSnapshot(state)
  return (
    <div>
      {snap.count}
      <button onClick={() => ++state.count}>+1</button>
    </div>
  )
}
```

**Comment by k11q** [comment]
hi @wobsoriano it seems you took down the repo, mind sharing how would you recommend I use valtio with solid, or if I dont need it? How would you approach state management with solid?

**Reply by wobsoriano** [reply]
Oh lmao what in the world. I used a tool to multi-delete repos and I might've accidentally included it 🤦🏼

Anyway I can republish the repo, I still have the source locally but I want it to return a signal instead of a solid proxy. In the meantime, you can check [solid-zustand](https://github.com/wobsoriano/solid-zustand) or [solid-jotai](https://github.com/wobsoriano/solid-jotai).

**Reply by wobsoriano** [reply]
Updated solid-jotai to 0.2.0.

I'll open the repo to public soon but for now, you can visit the npm package https://www.npmjs.com/package/solid-valtio

**Reply by k11q** [reply]
thanks! Looks great. @dai-shi would you consider supporting this solid binding and include it officially?

**Reply by dai-shi** [reply]
We encourage Valtio community to create third-party libraries and keep the core minimal.

**Reply by k11q** [reply]
I agree with that but I think bindings of other frameworks is essential in presenting this as a framework agnostic state management solution, which I think is one of its strongest point. Also having all the other framework bindings right in the framework helps build confidence for us to use for this exact purpose. Maybe you can consider it, also the community can help with developing them with prs and maintaining it, but directly integrated with the library instead of being third party

**Reply by mationai** [reply]
@wobsoriano Can you recreate the solid-valtio repo even if it doesn't return a signal? Having it available makes users feel better using it and it serves as a hub for issues/discussions. Others can add PR for signal if you don't have the time.

**Reply by wobsoriano** [reply]
> @wobsoriano Can you recreate the solid-valtio repo even if it doesn't return a signal? Having it available makes users feel better using it and it serves as a hub for issues/discussions. Others can add PR for signal if you don't have the time.

Let me check ok the weekend 🫡

**Reply by wobsoriano** [reply]
[Here you go.](https://github.com/wobsoriano/solid-valtio). You should be able to return a signal as well.

**Reply by wobsoriano** [reply]
If anyone can PR to make [promises + suspense](https://github.com/pmndrs/valtio#suspend-your-components) work, that'd be great and appreciated!
